src/app/client/src/app/modules/learn/components/course-consumption/course-player/course-player.component.ts

```diff
@@ -595,6 +604,10 @@ export class CoursePlayerComponent implements OnInit, OnDestroy {
           unit.isUnitConsumed = consumedContent.length === 1;
           unit.progress = consumedContent.length ? 100 : 0;
           unit.isUnitConsumptionStart = Boolean(consumedContent.length);
+          if(unit?.relationalMetadata?.optional=== false) {
```

Comment:
@Saurabhbalke please try to make it more concise using this if (!unit?.relationalMetadata?.optional)

Reply:
done